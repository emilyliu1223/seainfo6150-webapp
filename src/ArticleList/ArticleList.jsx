import React from 'react'
import ArticleListItem from "../ArticleListItem/ArticleListItem"
import styles from "./ArticleList.module.css";

const ArticleList = (props) => {
  let displayContent;
  console.log(props.articles)
  if (props.articles) {
    displayContent = (
      <div className={styles.container}>
        
        {props.articles.map((article) => (
          <ArticleListItem article={article} key={article.slug} />
        ))}
      </div>
      
    );
  } else {
    displayContent = <div>Try again</div>;
  }

  return (
    <div>
      {displayContent}
    </div>
  );
};


export default ArticleList
