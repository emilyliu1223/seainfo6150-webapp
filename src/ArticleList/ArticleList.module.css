.container {
    background: #999;
    border: 1px dashed #666;
    border-radius: 10px;
    box-shadow: 0px 0px 10px 5px rgba(204, 0, 0, 0.5);
    font-family: Arial, Helvetica, sans-serif;
    list-style-type: none;
    padding: 15px;
    width: fit-content;
    margin: auto;
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    grid-column-gap:50px;
    grid-row-gap: 50px;
    align-content: center;
    justify-content: space-between;
    
  

   
  }
  @media only screen and (max-width: 600px) {
    .container{
      grid-template-columns: repeat(1,1fr);
      grid-column-gap:10px;
    grid-row-gap: 10px;
    padding: 5px;
    }
  }
  @media only screen and (min-width: 600px) {
    .container{
      grid-template-columns: repeat(1,1fr);
      grid-column-gap:10px;
    grid-row-gap: 10px;
    padding: 5px;
    }
  }
  @media only screen and (min-width: 768px) {
    .container{
      grid-template-columns: repeat(2,1fr);
      grid-column-gap:20px;
    grid-row-gap: 20px;
    padding: 5px;
    }
  }
  @media only screen and (min-width: 992px) {
    .container{
      grid-template-columns: repeat(2,1fr);
      grid-column-gap:20px;
    grid-row-gap: 20px;
    padding: 5px;
    }
  }
  @media only screen and (min-width: 1200px) {
    .container{
      grid-template-columns: repeat(3,1fr);
    }
  }
  