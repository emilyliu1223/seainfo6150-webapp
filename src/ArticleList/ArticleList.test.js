
import ArticleList from "./ArticleList";

import React, {useEffect, useState} from "react";

import { isEmpty } from "lodash";


    
  
    
    
      const fetchData = async () => {
          
        // performs a GET request
        const response = await fetch(
          "http://demo1390455.mockable.io/articles"
        );
        const responseJson = await response.json();
        
      };
      console.log(fetchData);
     
      
   
  





describe("ArticleList tests", () => {
  it("renders correctly", () => {
    const { container } = render(<ArticleList articles={Object.values(fetchData)} />);
    expect(container).toMatchSnapshot();
  });
});