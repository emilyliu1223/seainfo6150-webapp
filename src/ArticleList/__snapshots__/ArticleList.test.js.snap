// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList tests renders correctly 1`] = `
<div>
  <div>
    <div
      class="container"
    />
  </div>
</div>
`;
